src/libstore/derivations.cc

```diff
@@ -1357,20 +1356,20 @@ nlohmann::json Derivation::toJSON(const StoreDirConfig & store) const
 
 Derivation Derivation::fromJSON(
     const StoreDirConfig & store,
-    const nlohmann::json & json,
+    const nlohmann::json & _json,
     const ExperimentalFeatureSettings & xpSettings)
 {
     using nlohmann::detail::value_t;
 
     Derivation res;
 
-    ensureType(json, value_t::object);
+    auto json = getObject(_json);
 
-    res.name = ensureType(valueAt(json, "name"), value_t::string);
+    res.name = getString(valueAt(json, "name"));
 
     try {
-        auto & outputsObj = ensureType(valueAt(json, "outputs"), value_t::object);
-        for (auto & [outputName, output] : outputsObj.items()) {
+        auto outputsObj = getObject(valueAt(json, "outputs"));
+        for (auto & [outputName, output] : outputsObj) {
```

Comment:
On the other hand, you could just inline `outputsObj` since it is just used in one place
```suggestion
        for (auto & [outputName, output] : getObject(valueAt(json, "outputs"))) {
```